g2p/cli.py

```diff
@@ -62,103 +136,185 @@ def cli():
 
-        You can list available mappings with "g2p doctor --list-ipa", or by visiting
-        http://g2p-studio.herokuapp.com/api/v1/langs .
+          To generate a mapping from IN_LANG-ipa to eng-ipa from a mapping following a
+          different patterns, e.g., from crl-equiv -> crl-ipa, specify both IN_LANG
+          (crl-equiv in this example) and OUT_LANG (crl-ipa in this example).
 
-        \b
-        Sample usage:
-            Generate Algonquin IPA to English IPA from alq -> alq-ipa
+          \b
+          Sample usage:
+            Generate Algonquin IPA to English IPA from alq -> alq-ipa:
                 g2p generate-mapping --ipa alq
-            Generate Mohawk IPA to English IPA from moh-equiv -> moh-ipa
+            Generate Mohawk IPA to English IPA from moh-equiv -> moh-ipa:
                 g2p generate-mapping --ipa moh-equiv moh-ipa
             Generate Michif IPA to English IPA from the union of crg-dv ->
-            crg-ipa and crg-tmd -> crg-ipa
+            crg-ipa and crg-tmd -> crg-ipa:
                 g2p generate-mapping --ipa --merge crg-dv:crg-tmd crg-ipa
-    """
 
-    if merge:
-        if out_lang is None:
-            raise click.BadParameter("OUT_LANG is required with --merge.")
-        in_langs = in_lang.split(":")
-    else:
-        in_langs = [in_lang]
+        List the dummy inventory used by --dummy:
+
+          g2p generate-mapping --list-dummy
+
+        From/to IPA mode:
+
+          g2p generate-mapping --from FROM_L1[:FROM_L2[:...]] --to TO_L1[:TO_L2[:...]]
+
+          Generate an IPA mapping from the union of FROM_L1-ipa, FROM-L2-ipa, etc to
+          the union of TO_L1-ipa, TO-L2-ipa, etc.
+
+          \b
+          Sample usage:
+            Generate a mapping from kwk-ipa to moh-ipa based on all mappings into
+            kwk-ipa and moh-ipa:
+                g2p generate-mapping --from kwk --to moh
+            Generate a mapping from eng-ipa to crg-ipa based only on crg-dv -> crg-ipa:
+                g2p generate-mapping --from eng --to crg-dv_to_crg-ipa
+
+          Full syntax for specifying FROM_Ln and TO_Ln:
+
+          \b
+            lang (i.e., 3-letter code):
+             - If there is only one mapping into lang-ipa, "lang" refers to the output
+               of that mapping, e.g., "fra" means "fra_to_fra-ipa,out".
+             - If there are several mappings into lang-ipa, "lang" refers to the union
+               of the outputs of those mappings, e.g., "moh" means the union of
+               "moh-equiv_to_moh-ipa,out" and "moh-festival_to_moh-ipa,out".
+             - It is an error if there are no mappings into lang-ipa.
+             - Only mappings from non-IPA to IPA are considered (i.e., IPA-to-IPA
+               mappings created by this command will not be included: use the longer
+               syntax below if you want to use them).
+             - Special case: "eng" refers to "eng-ipa_to_eng-arpabet,in".
+            in-lang_to_out-lang[,in|,out]:
```

Comment:
I just tried `g2p generate-mapping --from git --to eng,moh,str` and it failed - what am I doing wrong here? The CLI interpreted it as an ',in/,out' rule which I think is related to my gripe above:

`Error: Invalid value: Bad IPA lang spec "eng,moh,str": the ,in/,out option is only supported with the full in-lang_to_out-lang[,in][,out] syntax.`

Reply:
Yeah, you would need `g2p generate-mapping --from git --to eng:moh:str`. I agree with you, since we use `,` as a joiner elsewhere (for `-l` and `-o` in readalongs), it should be a joiner here too. You're not being annoying at all, this is an important gotcha that would have become an irritant if I didn't fix it.

So... 
 - `,` will now be an alternative joiner to `:`, so that `eng,moh,str`, `eng:moh,str` and `eng,moh,str` will all mean the same thing.
 - `[in/out]` is not bad, but not super either: if the file `in-lang_to_out-langi` exists, `in-lang_to_out-lang[in]` would be expanded by the shell to that file and would have to be quoted to prevent that. It's not a big deal, just a little gotcha. But I think it's going to be a sufficiently rare use case that I can live with that. So `[in]` and `[out]` it is.

Reply:
Hum, I see a problem. Since I use `[ ]` in the documentation to mean that something is optional, it will be confusing if suddenly it's meant to be used literally. I'd rather pick a different syntax.

Reply:
Would you be OK with a semicolon: `in-lang_to_out-lang;in` ?

Reply:
Code changed, now uses `[in]` and `[out]`.